builder/vmware/vmx/step_remote_upload.go

```diff
@@ -0,0 +1,87 @@
+package vmx
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	vmwcommon "github.com/hashicorp/packer/builder/vmware/common"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+// stepRemoteUpload uploads some thing from the state bag to a remote driver
+// (if it can) and stores that new remote path into the state bag.
+type stepRemoteUpload struct {
+	Key          string
+	Message      string
+	DoCleanup    bool
+	Checksum     string
+	ChecksumType string
+}
+
+func (s *stepRemoteUpload) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	driver := state.Get("driver").(vmwcommon.Driver)
+	ui := state.Get("ui").(packer.Ui)
+
+	remote, ok := driver.(vmwcommon.RemoteDriver)
+	if !ok {
+		return multistep.ActionContinue
+	}
+
+	path, ok := state.Get(s.Key).(string)
+	if !ok {
+		return multistep.ActionContinue
+	}
+
+	checksum, ok := state.Get(s.Checksum).(string)
```

Comment:
This isn't quite how the statebag works. If you're sending the floppy file checksum to the step struct, you don't need to get it out of the statebag -- you can simply access it directly using s.Checksum.

Reply:
Thanks @SwampDragons . Incorporated requested changes.